### PR TITLE
Remove dependency to matrix-auth plugin

### PR DESCRIPTION
**Pull request by mawinter69**
The only thing used from matrix-auth was the validation of user names.
Since matrix auth added a separation between users and groups the UI was
broken when matrix-auth 3.0 or newer was installed.

https://issues.jenkins.io/browse/JENKINS-68241
https://issues.jenkins.io/browse/JENKINS-67370
https://issues.jenkins.io/browse/JENKINS-67387
https://issues.jenkins.io/browse/JENKINS-67760
https://issues.jenkins.io/browse/JENKINS-67422
https://issues.jenkins.io/browse/JENKINS-67393

requires core 2.303.1 to make the icon stuff working

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java [hunk 1]
```diff
@@ -924,5 +939,62 @@ public boolean showPermission(String type, Permission p) {
...
+    private FormValidation doCheckName_(@NonNull String value, @NonNull Permission permission) {
+
+      final String v = value.substring(1,value.length()-1);
+      String ev = Functions.escape(v);
+
+      if (!Jenkins.get().hasPermission(permission))  return FormValidation.ok(ev); // can't check
+
+      SecurityRealm sr = Jenkins.get().getSecurityRealm();
+
+      if(v.equals("authenticated"))
```
Comment (AlexanderStohr) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java:956]:
```suggestion
      if (v.equals("authenticated")) {
```

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java [hunk 2]
```diff
@@ -924,5 +939,62 @@ public boolean showPermission(String type, Permission p) {
...
+        // system reserved group
+        return FormValidation.respond(FormValidation.Kind.OK, ValidationUtil.formatUserGroupValidationResponse("user", ev, "Group"));
+
```
Comment (AlexanderStohr) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java:959]:
```suggestion
    }
```

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/ValidationUtil.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  static String formatUserGroupValidationResponse(String img, String user, String tooltip) {
+      if (img == null) {
+          return String.format("<span title='%s'>%s</span>", tooltip, user);
+      }
+
+      if (jenkinsVersion.isOlderThan(new VersionNumber("2.308"))) {
+          return String.format("<span title='%s'><img src='%s%s/images/16x16/%s.png' style='margin-right:0.2em'>%s</span>", tooltip, Stapler.getCurrentRequest().getContextPath(), Jenkins.RESOURCE_PATH, img, user);
+      } else {
+          return String.format("<span title='%s'><img src='%s%s/images/svgs/%s.svg' width='16' style='margin-right:0.2em'>%s</span>", tooltip, Stapler.getCurrentRequest().getContextPath(), Jenkins.RESOURCE_PATH, img, user);
```
Comment (AlexanderStohr) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/ValidationUtil.java:27]: sort of code duplication, but joining both similar lines would mean an outer level of complexity,
such as defining
String image_format = "svgs/%s.svg' width='16"

**Comment by kuoruan** [comment]
@mawinter69 Thank you for the work, we really need this.
For thoses who need this fix: https://ci.jenkins.io/job/Plugins/job/role-strategy-plugin/view/change-requests/job/PR-172/

**Comment by SunBlack** [comment]
I tested your changes and it seems to work. But as you already doing more than this MR just should do (the icon is only one issue), could you also solve this (so we can use the artefacts of this MR until the maintainer starts to deploy a new version)?

image

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/ValidationUtil.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  static String formatUserGroupValidationResponse(String img, String user, String tooltip) {
+      if (img == null) {
+          return String.format("<span title='%s'>%s</span>", tooltip, user);
+      }
+
+      String imageFormat = String.format("svgs/%s.svg' width='16'", img);
+      if (jenkinsVersion.isOlderThan(new VersionNumber("2.308"))) {
+        imageFormat = String.format("16x16/%s.png'", img);
+      }
+      return String.format("<span title='%s'><img src='%s%s/images/%s style='margin-right:0.2em'>%s</span>", tooltip, Stapler.getCurrentRequest().getContextPath(), Jenkins.RESOURCE_PATH, imageFormat, user);
```
Comment (NotMyFault) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/ValidationUtil.java:28]: Preferably bump the baseline to 2.319.3 and use the icon API to access icons to get a rid of this check. These paths may be subject to future changes.

**Review by NotMyFault — COMMENTED** [review]
With the removal of the dependency of matrix-auth, I assume both plugins do still work alongside, as opposed to the PR title?

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/ValidationUtil.java (hunk 4 above)
Comment (mawinter69) [reply src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/ValidationUtil.java]: changed it so it uses the icon API without the need to go to higher baseline

Comment (NotMyFault) [reply src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/ValidationUtil.java]: Thanks, do you mind elaborating on my concern above? I'm not a maintainer of this plugin, but I wouldn't mind cutting another release including this PR, if it's ready for merge.

**Comment by mawinter69** [comment]
both matrix-auth and rolestrategy are authorization plugins, so you can effectively use only one at a time. But you can install both in parallel. 
So with this change you can now install rolestrategy without the need to install matrix-auth.

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**